File: unique-morse-code-words/unique-morse-code-words.py
class Solution:
    def uniqueMorseRepresentations(self, words: List[str]) -> int:
        morse_list =[".-","-...","-.-.","-..",".","..-.","--.","....","..",".---","-.-",".-..","--","-.","---",".--.","--.-",".-.","...","-","..-","...-",".--","-..-","-.--","--.."]
        morse_dict = {}
        for i in range(len(morse_list)):
            morse_dict[chr(i+97)] = morse_list[i]
        
        res_set = set()
        for i in words:
            res_str = ""
            for j in i:
                res_str += morse_dict[j]
            res_set.add(res_str)
        return len(res_set)

File: unique-morse-code-words/unique-morse-code-words.go
func uniqueMorseRepresentations(words []string) int {
    morse_list := [26]string{".-","-...","-.-.","-..",".","..-.","--.","....","..",".---","-.-",".-..","--","-.","---",".--.","--.-",".-.","...","-","..-","...-",".--","-..-","-.--","--.."}
    
    char_list := make(map[string]string)
    
    for i:=0;i<26;i++{
        char_list[string(i+97)] = morse_list[i]
    }
    
    res_str := ""
    res_list := make(map[string]string)
    for i:=0;i<len(words);i++{
        res_str = ""
        for j:=0;j<len(words[i]);j++{
            res_str = res_str + char_list[string(words[i][j])]
        }
        res_list[res_str] = "abc"
    }
    
    return len(res_list)
}